### ISSUE TITLE
Suggestion query comparison and history

### ISSUE THREAD

**Issue by donjerick** [MiguelCatalan/MaterialSearchView#165]
Would it be better if SearchAdapter compares each word in query instead of using startsWith()?

Current code in SearchAdapter.java
`if (string.toLowerCase().startsWith(constraint.toString().toLowerCase())) {
    searchData.add(string);
}`

Also would it be useful to have a method to save history searches. Adding the history search on top of the suggestions will be convenient for users.

**Comment by Kevuardo** [comment]
I agree totally with you. Right now I'm struggling with that and it's something that could be improved, both searching method and history.

Also, I would rather use a method that instead of searching for words starting with what you typed as query, it would search for matches in strings, just like splitting the query string, looking for a match of first letter, and if found look for one of the second, and so on.

But I guess we will have to do it ourselves.

**Comment by LeonardoEmili** [comment]
Since I'm reading now your post I answer here. Still this functionality is not present, I successfully managed to add it by providing the searchview a custom Adapter. Inside it I modified the Filter method by changing the filter logic. I also added history to my searchview combinating the power of SharedPreferences for storing user researches and changing the ImageView drawable if the entry has ever been searched.
